executor/runtime/docker/docker_ssh.go

```diff
@@ -157,3 +168,28 @@ func addContainerSSHDConfigWithData(c runtimeTypes.Container, tw *tar.Writer, cf
 	}
 	return nil
 }
+
+func getContainerSSHPolicyHash(c runtimeTypes.Container) (string, error) {
+	metatronCreds := c.MetatronCreds()
+	if metatronCreds == nil {
+		return "", nil
+	}
+	appMetadata := metatronCreds.AppMetadata
+	if appMetadata == nil && *appMetadata == "" {
+		return "", nil
+	}
+	params, err := url.ParseQuery(*appMetadata)
+	if err != nil {
+		return "", fmt.Errorf("Failed to parse metatron app metadata: %v", err)
+	}
+	sshPolicy := params.Get("sshPolicy")
+	if sshPolicy == "" {
+		return "", nil
+	}
+	h := sha256.New()
+	_, err = h.Write([]byte(sshPolicy))
+	if err != nil {
+		return "", fmt.Errorf("Failed to hash ssh policy: %v", err)
+	}
+	return base64.StdEncoding.EncodeToString(h.Sum(nil)), nil
```

Comment:
Wait, why do we need to base64 a shasum? It should be ascii-safe?

Reply:
Yea, this is being added to the authorized principals file. I could make this a bit more clear and return the []byte hash here and do the encoding when it's inserted into the authorized principals.